pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,369 @@
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.ClassifierStage: {
+					ClassifierTable,
+				},
+				binding.ValidationStage: {
+					SpoofGuardTable,
+				},
+				binding.ConntrackStateStage: {
+					ConntrackTable,
+					ConntrackStateTable,
+				},
+				binding.RoutingStage: {
+					L3ForwardingTable,
+					L3DecTTLTable,
+				},
+				binding.SwitchingStage: {
+					L2ForwardingCalcTable,
+				},
+				binding.ConntrackStage: {
+					ConntrackCommitTable,
+				},
+				binding.OutputStage: {
+					L2ForwardingOutTable,
+				},
+			},
+		}
+		if c.enableMulticast {
+			template.addTable(binding.ValidationStage, MulticastClassifierTable)
+		}
+		for _, ipProtocol := range c.ipProtocols {
+			if ipProtocol == binding.ProtocolIPv6 {
+				template.addTable(binding.ValidationStage, IPv6Table)
+				break
+			}
+		}
+	case pipelineARP:
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.ValidationStage: {
+					ARPSpoofGuardTable,
+				},
+				binding.OutputStage: {
+					ARPResponderTable,
+				},
+			},
+		}
+	}
+
+	return template
+}
+
+func (c *featureNetworkPolicy) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineIP {
+		return nil
+	}
+	template := &featureTemplate{
+		stageTables: map[binding.StageID][]*FeatureTable{
+			binding.EgressSecurityStage: {
+				EgressRuleTable,
+				EgressDefaultTable,
+				EgressMetricTable,
+			},
+			binding.IngressSecurityStage: {
+				IngressSecurityClassifierTable,
+				IngressRuleTable,
+				IngressDefaultTable,
+				IngressMetricTable,
+			},
+		},
+	}
+	if c.enableAntreaPolicy {
+		template.addTable(binding.EgressSecurityStage, AntreaPolicyEgressRuleTable)
+		template.addTable(binding.IngressSecurityStage, AntreaPolicyIngressRuleTable)
+	}
+	return template
+}
+
+func (c *featureService) getTemplate(p pipeline) *featureTemplate {
+	if p != pipelineIP {
+		return nil
+	}
+	var template *featureTemplate
+	if c.enableProxy {
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.ConntrackStateStage: {
+					SNATConntrackTable,
+				},
+				binding.PreRoutingStage: {
+					PreRoutingClassifierTable,
+					SessionAffinityTable,
+					ServiceLBTable,
+					EndpointDNATTable,
+				},
+				binding.RoutingStage: {
+					L3ForwardingTable,
+					ServiceHairpinMarkTable,
+				},
+				binding.PostRoutingStage: {
+					SNATConntrackCommitTable,
+				},
+				binding.ConntrackStage: {
+					ConntrackCommitTable,
+				},
+				binding.OutputStage: {
+					L2ForwardingOutTable,
+				},
+			},
+		}
+		if runtime.IsWindowsPlatform() {
+			template.addTable(binding.ClassifierStage, UplinkTable)
+		}
+		if c.proxyAll {
+			template.addTable(binding.PreRoutingStage, NodePortProbeTable)
+		}
+	} else {
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.PreRoutingStage: {
+					DNATTable,
+				},
+			},
+		}
+	}
+	return template
```

Comment:
you can move this else to top like below to reduce space indent.
```
	if !c.enableProxy {
		template = &featureTemplate{
			stageTables: map[binding.StageID][]*FeatureTable{
				binding.PreRoutingStage: {
					DNATTable,
				},
			},
		}
		return template
	}
  
}
```

Reply:
Sorry, I didn't find out what's the difference. Is the space indent in code above reduced?

Reply:
it use `!c.enableProxy` to check the non-proxy case first. the full codes should be like below:
```
	if !c.enableProxy {
		template = &featureTemplate{
			stageTables: map[binding.StageID][]*FeatureTable{
				binding.PreRoutingStage: {
					DNATTable,
				},
			},
		}
		return template
	}
	template = &featureTemplate{
		stageTables: map[binding.StageID][]*FeatureTable{
			binding.ConntrackStateStage: {
				SNATConntrackTable,
			},
			binding.PreRoutingStage: {
				PreRoutingClassifierTable,
				SessionAffinityTable,
				ServiceLBTable,
				EndpointDNATTable,
			},
			binding.RoutingStage: {
				L3ForwardingTable,
				ServiceHairpinMarkTable,
			},
			binding.PostRoutingStage: {
				SNATConntrackCommitTable,
			},
			binding.ConntrackStage: {
				ConntrackCommitTable,
			},
			binding.OutputStage: {
				L2ForwardingOutTable,
			},
		},
	}
	if runtime.IsWindowsPlatform() {
		template.addTable(binding.ClassifierStage, UplinkTable)
	}
	if c.proxyAll {
		template.addTable(binding.PreRoutingStage, NodePortProbeTable)
	}
	return template
```